src/openforms/submissions/tasks/__init__.py

```diff
@@ -22,127 +21,50 @@
 logger = logging.getLogger(__name__)
 
 
-def on_completion(submission_id: int) -> None:
-    """
-    Celery chain of tasks to execute on a submission completion event.
+def on_post_submission_event(submission_id: int) -> None:
+    # this can run any time because they have been claimed earlier
+    cleanup_temporary_files_for.delay(submission_id)
 
-    This SHOULD be invoked as a transaction.on_commit(...) handler, therefore it should
-    not execute any extra queries in the process this function is running in.
-    """
-    # use immutable signatures so that the result of previous tasks is not passed
-    # in as an argument to chained tasks
+    # If the form involves appointments and no appointment has been scheduled yet, schedule it.
+    # Todo: deprecated => Not needed with the new appointment flow
     register_appointment_task = maybe_register_appointment.si(submission_id)
```

Comment:
You can add a `warnings.warn("...", PendingDeprecationWarning)` in the `maybe_register_appointment` task too if you want.